enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -0,0 +1,447 @@
+---
+title: point-to-point-network-check
+authors:
+  - "@deads2k"
+  - "@sanchezl"
+reviewers:
+approvers:
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+replaces:
+superseded-by:
+---
+
+# Stability: Point to Point Network Check
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+There are a few key network connections that are important to the health of the cluster.
+We can create a CRD to store the the last success time for each point-to-point check.
+By watching for explicit failures and lack of updates, we can produce a picture of when a point-to-point check failed.
+This information could be correlated to e2e failures, alerts, events, logs, or other information feeds.
+
+## Motivation
+
+This attempts to identify failures of particular connections in a black-box manner that is independent of SDN, cloud,
+or cluster ingress.
+
+This enhancement compliments existing metrics reported by REST client instrumentation. This enhancement derives
+additional value by:
+* REST calls can fail for a variety of reasons beyond simple DNS lookup and reachability of the target endpoint.
+* Limiting scope to DNS lookup and TCP connect to target endpoint. REST client can fail for a variety of other reasons.
+* Providing insights into target endpoints that are not accessed a REST client.
+* Detecting and reporting issues independently of monitoring, which might itself not be available.
+* Providing insights during a rollout.
+* Providing status that can be consumed by an operator.
+
+### Goals
+
+1. be independent of SDN, cloud, or cluster ingress
```

Comment:
SDN surprised me here. Pod IPs will be SDN dependent. If you mean that the recording of the test results should be independent, be more clear here.